File: modification.c
//Write a program to perform modification of an element/node in a given list.

#include <stdio.h>
#include <stdlib.h>

// Node structure
struct Node {
    int data;
    struct Node* next;
};

struct Node* head = NULL;

// Create and insert at end
void insert(int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;

    if (head == NULL) {
        head = newNode;
        return;
    }

    struct Node* temp = head;
    while (temp->next != NULL)
        temp = temp->next;

    temp->next = newNode;
}

// Modify a node's data
void modify(int oldValue, int newValue) {
    struct Node* temp = head;
    while (temp != NULL) {
        if (temp->data == oldValue) {
            temp->data = newValue;
            printf("Node with value %d modified to %d.\n", oldValue, newValue);
            return;
        }
        temp = temp->next;
    }
    printf("Value %d not found in the list.\n", oldValue);
}

// Display the list
void display() {
    struct Node* temp = head;
    printf("List: ");
    while (temp != NULL) {
        printf("%d -> ", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}

int main() {
    int choice, val, oldVal, newVal;

    while (1) {
        printf("\n1. Insert\n2. Modify\n3. Display\n4. Exit\n");
        printf("Enter choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter value to insert: ");
                scanf("%d", &val);
                insert(val);
                break;
            case 2:
                printf("Enter value to modify: ");
                scanf("%d", &oldVal);
                printf("Enter new value: ");
                scanf("%d", &newVal);
                modify(oldVal, newVal);
                break;
            case 3:
                display();
                break;
            case 4:
                exit(0);
        }
    }
    return 0;
}

File: modification.py
#Write a program to perform modification of an element/node in a given list.

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert(self, value):
        new_node = Node(value)
        if not self.head:
            self.head = new_node
            return
        temp = self.head
        while temp.next:
            temp = temp.next
        temp.next = new_node

    def modify(self, old_value, new_value):
        temp = self.head
        while temp:
            if temp.data == old_value:
                temp.data = new_value
                print(f"Modified node value from {old_value} to {new_value}.")
                return
            temp = temp.next
        print(f"Value {old_value} not found in list.")

    def display(self):
        temp = self.head
        print("List:", end=" ")
        while temp:
            print(f"{temp.data} -> ", end="")
            temp = temp.next
        print("NULL")

# Example usage
ll = LinkedList()

while True:
    print("\n1. Insert\n2. Modify\n3. Display\n4. Exit")
    choice = int(input("Enter your choice: "))
    if choice == 1:
        val = int(input("Enter value to insert: "))
        ll.insert(val)
    elif choice == 2:
        old = int(input("Enter value to modify: "))
        new = int(input("Enter new value: "))
        ll.modify(old, new)
    elif choice == 3:
        ll.display()
    elif choice == 4:
        break
    else:
        print("Invalid choice!")
